From 84bc203529c8f355308f13defe1b86e862f0ce0d Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Fri, 29 Oct 2021 12:51:55 +0200
Subject: [PATCH] Fix tests on aarch64 and mips architecture

The divide_by_zero is meant to run if both __aarch64__ and __mips__ are not defined,
meaning that the test is being compiled on any other architecture.
---
 test/suicide.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/suicide.cpp b/test/suicide.cpp
index aa8022c..e694c6f 100644
--- a/test/suicide.cpp
+++ b/test/suicide.cpp
@@ -58,7 +58,7 @@ void abort_abort_I_repeat_abort_abort() {
 TEST_ABORT(calling_abort) { abort_abort_I_repeat_abort_abort(); }
 
 // aarch64 and mips does not trap Division by zero
-#if !defined(__aarch64__) || !defined(__mips__)
+#if !defined(__aarch64__) && !defined(__mips__)
 volatile int zero = 0;
 
 int divide_by_zero() {
